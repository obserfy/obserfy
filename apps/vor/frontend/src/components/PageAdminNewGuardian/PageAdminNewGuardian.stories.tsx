import { FC } from "react"
import PageAdminNewGuardian from "./PageAdminNewGuardian"

export default {
  title: "Core/PageAdminNewGuardian",
  component: PageAdminNewGuardian,
  parameters: {
    componentSubtitle: "Just a simple PageAdminNewGuardian",
  },
}

export const Basic: FC = () => <PageAdminNewGuardian />
