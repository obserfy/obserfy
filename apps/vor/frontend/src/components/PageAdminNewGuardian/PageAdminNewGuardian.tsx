import { t, Trans } from "@lingui/macro"
import { FC, useState } from "react"
import { Box, Button } from "theme-ui"
import { usePostNewGuardian } from "../../hooks/api/guardians/usePostNewGuardian"
import { ADMIN_GUARDIAN_URL } from "../../routes"
import BackNavigation from "../BackNavigation/BackNavigation"
import Input from "../Input/Input"
import { navigate } from "../Link/Link"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"
import TextArea from "../TextArea/TextArea"
import { Typography } from "../Typography/Typography"

export const PageAdminNewGuardian: FC = () => {
  const [name, setName] = useState("")
  const [email, setEmail] = useState("")
  const [phone, setPhone] = useState("")
  const [note, setNote] = useState("")

  const { mutateAsync, status } = usePostNewGuardian()

  return (
    <Box mx="auto" sx={{ maxWidth: "maxWidth.sm" }}>
      <BackNavigation to={ADMIN_GUARDIAN_URL} text="All Guardians" />
      <Typography.H5 mx={3} mb={3}>
        <Trans>New Guardian</Trans>
      </Typography.H5>
      <Box p={3}>
        <Input
          value={name}
          mb={2}
          label={t`Guardian Name`}
          sx={{ width: "100%" }}
          onChange={(e) => setName(e.target.value)}
        />
        <Input
          type="email"
          value={email}
          mb={2}
          label={t`Email`}
          sx={{ width: "100%" }}
          onChange={(event) => setEmail(event.target.value)}
        />
        <Input
          type="phone"
          value={phone}
          mb={3}
          label={t`Phone`}
          sx={{ width: "100%" }}
          onChange={(event) => setPhone(event.target.value)}
        />
        <TextArea
          mb={3}
          label={t`Note`}
          value={note}
          onChange={(e) => setNote(e.target.value)}
        />
        <Button
          sx={{ width: "100%" }}
          disabled={name === ""}
          onClick={async () => {
            try {
              await mutateAsync({
                email,
                name,
                phone,
                note,
              })
              await navigate(ADMIN_GUARDIAN_URL)
            } catch (e) {
              Sentry.captureException(e)
            }
          }}
        >
          {status === "loading" && <LoadingIndicator color="onPrimary" />}
          <Trans>Save</Trans>
        </Button>
      </Box>
    </Box>
  )
}

export default PageAdminNewGuardian
