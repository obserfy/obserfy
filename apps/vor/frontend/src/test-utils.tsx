/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
