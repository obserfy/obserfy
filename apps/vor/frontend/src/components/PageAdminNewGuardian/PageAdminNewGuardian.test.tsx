import { render } from "../../test-utils"
import { Basic } from "./PageAdminNewGuardian.stories"

describe("PageAdminNewGuardian", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
