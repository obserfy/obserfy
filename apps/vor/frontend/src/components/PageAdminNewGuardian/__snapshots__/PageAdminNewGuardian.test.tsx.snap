// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageAdminNewGuardian should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 640px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin: 16px;
  margin-right: 4px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
  margin-left: 16px;
  margin-right: 16px;
  margin-bottom: 16px;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 4px;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 8px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-10:autofill,
.emotion-10:autofill:hover,
.emotion-10:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-10:autofill:first-line,
.emotion-10:autofill:hover:first-line,
.emotion-10:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-10:-webkit-autofill,
.emotion-10:-webkit-autofill:hover,
.emotion-10:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-10:-webkit-autofill:first-line,
.emotion-10:-webkit-autofill:hover:first-line,
.emotion-10:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-10:hover,
.emotion-10:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-10::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-10::-moz-placeholder {
  font-style: italic;
}

.emotion-10:-ms-input-placeholder {
  font-style: italic;
}

.emotion-10::placeholder {
  font-style: italic;
}

.emotion-10:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-18:autofill,
.emotion-18:autofill:hover,
.emotion-18:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-18:autofill:first-line,
.emotion-18:autofill:hover:first-line,
.emotion-18:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-18:-webkit-autofill,
.emotion-18:-webkit-autofill:hover,
.emotion-18:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-18:-webkit-autofill:first-line,
.emotion-18:-webkit-autofill:hover:first-line,
.emotion-18:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-18:hover,
.emotion-18:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-18::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-18::-moz-placeholder {
  font-style: italic;
}

.emotion-18:-ms-input-placeholder {
  font-style: italic;
}

.emotion-18::placeholder {
  font-style: italic;
}

.emotion-18:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 16px;
  height: 180px;
  resize: vertical;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
}

.emotion-21:hover,
.emotion-21:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-21::-webkit-input-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-21::-moz-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-21:-ms-input-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-21::placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  width: 100%;
}

.emotion-22:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-22:hover,
.emotion-22:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-22:hover:disabled,
.emotion-22:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        href="/dashboard/admin/guardians"
        style="display: block;"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-3"
          />
          <p
            class="emotion-4"
          >
            All Guardians
          </p>
        </div>
      </a>
      <h5
        class="emotion-5"
      >
        New Guardian
      </h5>
      <div
        class="emotion-6"
      >
        <label
          class="emotion-7"
        >
          <div
            class="emotion-8"
          >
            Guardian Name
          </div>
          <div
            class="emotion-9"
          >
            <input
              class="emotion-10"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-7"
        >
          <div
            class="emotion-8"
          >
            Email
          </div>
          <div
            class="emotion-9"
          >
            <input
              class="emotion-10"
              type="email"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-7"
        >
          <div
            class="emotion-8"
          >
            Phone
          </div>
          <div
            class="emotion-9"
          >
            <input
              class="emotion-18"
              type="phone"
              value=""
            />
          </div>
        </label>
        <label
          class="emotion-19"
        >
          <div
            class="emotion-8"
          >
            Note
          </div>
          <textarea
            aria-label="Note"
            class="emotion-21"
          />
        </label>
        <button
          class="emotion-22"
          disabled=""
        >
          Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
